charts/external-dns/templates/deployment.yaml

```diff
@@ -132,6 +136,47 @@ spec:
           resources:
             {{- toYaml . | nindent 12 }}
           {{- end }}
+        {{- if eq $provider "webhook" }}
+        {{- with .Values.provider }}
+        - name: {{ $providername }}
+          image: {{ get $whp .name }}
+          imagePullPolicy: {{ $.Values.image.pullPolicy }}
+          {{- with .env }}
+          env:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          {{- with .args }}
+          args:
+            - {{ toYaml . | nindent 12 }}
+          {{- end }}
+          ports:
+            - name: http
+              protocol: TCP
+              containerPort: 8888
+          livenessProbe:
+            {{- toYaml $.Values.livenessProbe | nindent 12 }}
+          readinessProbe:
+            {{- toYaml $.Values.readinessProbe | nindent 12 }}
+          {{- if or $.Values.secretConfiguration.enabled $.Values.extraVolumeMounts }}
+          volumeMounts:
+            {{- if $.Values.secretConfiguration.enabled }}
+            - name: secrets
+              mountPath: {{ tpl $.Values.secretConfiguration.mountPath $ }}
```

Comment:
Secrets should not be shared across the main container and sidecar. They should only be readable to containers that need them in order to reduce the risk of compromise. Most of the time (when running a webhook provider) only the webhook sidecar will need secrets, though there are a few obscure sources that could need secrets.

Reply:
> Most of the time (when running a webhook provider) only the webhook sidecar will need secrets.

Oh That's right ! 
:thinking: Wdyt then about mounting the secret only on external-dns for in-tree provider and only on the sidecar for webhook providers ?
cc @stevehipwell @mrueg

Reply:
I would be strongly in favour of not supporting `secretConfiguration` for the webhook sidecar as secrets should preferably be managed outside of the Helm chart and then interacted with via the `extraVolumes` & `extraVolumeMounts` values.

Reply:
The `cloudfoundry` source takes a password, but I suppose that can only come in through args, not a mounted secret. So perhaps we could say it would be unlikely for a future source to require a mounted secret.

I wouldn't want to have `secretConfiguration` only work for in-tree providers. That would be an odd discontinuity.

Reply:
I added specific extraVolumeMounts for sidecar with 91095975

Reply:
I still don't think we should be explicitly using `secretConfiguration` here, the secret _COULD_ be mounted by the `extraVolumeMounts` but we should be recommending external secrets be used if needed.